client/file-manager/src/rocksdb.rs

```diff
@@ -0,0 +1,1023 @@
+            .map_err(|_| FileStorageError::FailedToParseFileMetadata)?;
+        let raw_root = metadata.fingerprint.as_ref();
+        let mut root = convert_raw_bytes_to_hasher_out::<T>(raw_root.to_vec())
+            .map_err(|_| FileStorageError::FailedToParseFingerprint)?;
+        let file_trie = RocksDbFileDataTrie::<T>::from_existing(self.storage.clone(), &mut root);
+
+        file_trie.get_chunk(chunk_id)
+    }
+
+    fn write_chunk(
+        &mut self,
+        key: &HasherOutT<T>,
+        chunk_id: &ChunkId,
+        data: &Chunk,
+    ) -> Result<FileStorageWriteOutcome, FileStorageWriteError> {
+        let raw_metadata = self
+            .storage
+            .read()
+            .expect("Failed to acquire read lock")
+            .read(*key)
+            .map_err(|_| FileStorageWriteError::FailedToReadStorage)?
+            .expect("Failed to find File Metadata");
+
+        let metadata: FileMetadata = serde_json::from_slice(&raw_metadata)
+            .map_err(|_| FileStorageWriteError::FailedToParseFileMetadata)?;
+        let raw_root = metadata.fingerprint.as_ref();
+        let mut root = convert_raw_bytes_to_hasher_out::<T>(raw_root.to_vec())
+            .map_err(|_| FileStorageWriteError::FailedToParseFingerprint)?;
+        let mut file_trie =
+            RocksDbFileDataTrie::<T>::from_existing(self.storage.clone(), &mut root);
+
+        file_trie
+            .write_chunk(chunk_id, data)
+            .map_err(|_| FileStorageWriteError::FailedToInsertFileChunk)?;
+
+        Ok(FileStorageWriteOutcome::FileComplete)
+    }
+
+    fn insert_file(
+        &mut self,
+        key: HasherOutT<T>,
+        metadata: FileMetadata,
+    ) -> Result<(), FileStorageError> {
+        let mut transaction = DBTransaction::new();
+        let serialized_metadata = serde_json::to_vec(&metadata)
+            .map_err(|_| FileStorageError::FailedToParseFileMetadata)?;
+
+        transaction.put(0, key.as_ref(), &serialized_metadata);
+
+        self.storage
+            .write()
+            .expect("Failed to acquire write lock")
+            .write(transaction)
+            .map_err(|_| FileStorageError::FailedToWriteToStorage)?;
+
+        Ok(())
+    }
+
+    fn insert_file_with_data(
+        &mut self,
+        key: HasherOutT<T>,
+        metadata: FileMetadata,
+        _file_data: Self::FileDataTrie,
+    ) -> Result<(), FileStorageError> {
+        let mut transaction = DBTransaction::new();
+        let raw_metadata = serde_json::to_vec(&metadata)
+            .map_err(|_| FileStorageError::FailedToParseFileMetadata)?;
+
+        transaction.put(0, key.as_ref(), &raw_metadata);
+
+        self.storage
+            .write()
+            .expect("Failed to acquire write lock")
+            .write(transaction)
+            .map_err(|_| FileStorageError::FailedToWriteToStorage)?;
+
+        Ok(())
+    }
+
+    fn get_metadata(&self, key: &HasherOutT<T>) -> Result<FileMetadata, FileStorageError> {
```

Comment:
File manager is bypassing file trie. There should be one way of accessing and writing data to storage.

Reply:
I'm gonna remove the storage from File Storage in the next PR